contracts/protocol/modules/v1/RebasingComponentAssetLimitModule.sol

```diff
@@ -0,0 +1,331 @@
+import { INAVIssuanceHook } from "../../../interfaces/INAVIssuanceHook.sol";
+import { ISetToken } from "../../../interfaces/ISetToken.sol";
+import { ModuleBase } from "../../lib/ModuleBase.sol";
+
+/**
+ * @title RebasingComponentAssetLimitModule
+ * @author Index Coop
+ * @notice NAVIssuanceModule hook that checks the issue and redeem amounts are lower than a given limit 
+ * and syncs rebasing components position.
+ */
+contract RebasingComponentAssetLimitModule is ModuleBase, ReentrancyGuard, Ownable, INAVIssuanceHook {
+    using AddressArrayUtils for address[];
+
+    /* ============ Events ============ */
+
+    /**
+     * @dev Emitted when rebasing components are added or removed.
+     * @param _setToken       Instance of SetToken whose rebasing components are updated.
+     * @param _isAdded        True if components are added, false if removed.
+     * @param _components     Array of rebasing components being added/removed.
+     */
+    event RebasingComponentsUpdated(
+        ISetToken indexed _setToken,
+        bool indexed _isAdded,
+        IERC20[] _components
+    );
+
+    /* ============ State Variables ============ */
+
+    // Mapping of asset to its limit
+    mapping(ISetToken => mapping(address => uint256)) assetLimits;
+
+    // Array of assets that have limits
+    mapping(ISetToken => address[]) public assets;
+
+    // Mapping to efficiently check if rebasing component is enabled in SetToken
+    mapping(ISetToken => mapping(IERC20 => bool)) public rebasingComponentEnabled;
+
+    // Internal mapping of enabled rebasing components for syncing positions
+    mapping(ISetToken => address[]) internal rebasingComponents;
+
+    /* ============ Constructor ============ */
+
+    /**
+     * @dev Initializes the module with the controller address.
+     * @param _controller  Address of the controller contract.
+     */
+    constructor(IController _controller) public ModuleBase(_controller) { }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * @dev CALLABLE BY ANYBODY: Sync Set positions with ALL enabled rebasing component positions.
+     * @param _setToken    Instance of the SetToken
+     */
+    function sync(ISetToken _setToken) public nonReentrant onlyValidAndInitializedSet(_setToken) {
+        uint256 setTotalSupply = _setToken.totalSupply();
+
+        // Only sync positions when Set supply is not 0. Without this check, if sync is called by someone before the
+        // first issuance, then editDefaultPosition would remove the default positions from the SetToken
+        if (setTotalSupply > 0) {
+            address[] memory setTokenRebasingComponents = rebasingComponents[_setToken];
+            for (uint256 i = 0; i < setTokenRebasingComponents.length; i++) {
+                IERC20 component = IERC20(setTokenRebasingComponents[i]);
+
+                uint256 previousPositionUnit = _setToken.getDefaultPositionRealUnit(address(component)).toUint256();
```

Comment:
external rebasing positions usually need specific modules, which is where the position issuance hook logic can be implemented (i.e., `AaveV3LeverageModule` & `sync`)

this module targets rebasing erc20 positions which may otherwise not need specific modules 

open to any suggestions to make this more clear